test/test_formatting.py

```diff
@@ -3,1177 +3,1163 @@
+    ('RESET_SPACE', action(space_char='ABCD'),
+     action()),
+    # RESET_CASE: No case
+    ('RESET_CASE', action(case=action().CASE_UPPER),
+     action()),
+    # SET_SPACE:xy: Set space to xy
+    ('SET_SPACE:', action(space_char='test'),
+     action(space_char='')),
+    ('SET_SPACE:-', action(space_char='test'),
+     action(space_char='-')),
+    ('SET_SPACE:123 45', action(space_char='test'),
+     action(space_char='123 45')),
+    # RESET: No case, default space
+)
+
+@parametrize(CHANGE_MODE_TESTS)
+def test_change_mode(command, action, expected_action):
+    assert formatting._change_mode(command, action) == expected_action
+
+
+last_action_normal = action()
+last_action_capitalized = action(capitalize=True)
+last_action_attached = action(attach=True)
+
+META_CARRY_CAPITALIZE_TESTS = (
+
+    # Test word handling and space handling, standard.
+    ('~|*', last_action_normal, spaces_before,
+     (action(word='*', text=' *'))),
+    ('~|*', last_action_normal, spaces_after,
+     (action(word='*', text='* '))),
+
+    # With attach flags:
+    ('~|*^', last_action_normal, spaces_before,
+     (action(word='*', text=' *', attach=True))),
+    ('~|*^', last_action_normal, spaces_after,
+     (action(word='*', text='*', attach=True))),
+    ('^~|*', last_action_normal, spaces_before,
+     (action(word='*', text='*'))),
+    ('^~|*', last_action_normal, spaces_after,
+     (action(word='*', text='* '))),
+    ('^~|*^', last_action_normal, spaces_before,
+     (action(word='*', text='*', attach=True))),
+    ('^~|*^', last_action_normal, spaces_after,
+     (action(word='*', text='*', attach=True))),
+
+    # Should 'do nothing'.
+    ('~|', last_action_capitalized, spaces_before,
+     (last_action_capitalized)),
+
+    # Should lose 'attach' flag.
+    ('~|', last_action_attached, spaces_before,
+     (action())),
+
+    # Verify capitalize carry.
+    ('^~|^', last_action_capitalized, spaces_before,
+     (action(capitalize=True, attach=True))),
+    ('^~|aset^', last_action_capitalized, spaces_before,
+     (action(capitalize=True, attach=True, word="aset", text="aset"))),
+    ('~|aset', last_action_capitalized, spaces_before,
+     (action(capitalize=True, word="aset", text=" aset"))),
+    ('~|aset', last_action_capitalized, spaces_after,
+     (action(capitalize=True, word="aset", text="aset "))),
+
+    # Verify 'attach' flag overriding.
+    ('~|aset', last_action_attached, spaces_after,
+     (action(word="aset", text="aset "))),
+    ('~|aset^', last_action_attached, spaces_after,
+     (action(word="aset", text="aset", attach=True))),
+
+)
+
+@parametrize(META_CARRY_CAPITALIZE_TESTS)
+def test_meta_carry_capitalize(meta, last_action, spaces_after, expected):
+    assert formatting._apply_carry_capitalize(meta, last_action, spaces_after) == expected
+
+
+class TestApplyCase(object):
+
+    test = ' some test '
+    test2 = 'test Me'
+    test3 = ' SOME TEST '
+
+    TESTS = (
+        # UNDEFINED
+        (test, '', Fa
```

Comment:
Can we change the spelling to "optimization" because most of the code base is in American spelling?

Reply:
Done.